app/webhook.go

```diff
@@ -145,15 +150,27 @@ func (a *App) TriggerWebhook(payload *model.OutgoingWebhookPayload, hook *model.
 	}
 }
 
-func (a *App) doOutgoingWebhookRequest(url string, body io.Reader, contentType string) (*model.OutgoingWebhookResponse, error) {
-	req, err := http.NewRequest("POST", url, body)
+func (a *App) doOutgoingWebhookRequest(url string, body io.Reader, contentType, secret string) (*model.OutgoingWebhookResponse, error) {
+	bodyBytes, err := ioutil.ReadAll(body)
+	if err != nil {
+		return nil, errors.New("OutgoingWehookRequest payload failed to read.")
+	}
+	bodyReader := bytes.NewReader(bodyBytes)
+
+	req, err := http.NewRequest("POST", url, bodyReader)
 	if err != nil {
 		return nil, err
 	}
 
 	req.Header.Set("Content-Type", contentType)
 	req.Header.Set("Accept", "application/json")
 
+	nowStamp := strconv.FormatInt(time.Now().Unix(), 10)
+	// nowStamp = "1576740798" // for testing only
+	hmacDigest := model.GenerateHmacSignature(&bodyBytes, nowStamp, secret)
```

Comment:
`GenerateHmacSignature` seems like it should take a `time.Time`, and we only render to an integer when setting the header.

Reply:
We can do that. Hmac needs bytes to sign with so we need the integer in string whether it's done in GenerateSignature or like this. Also it seems like HTTP takes everything in header as string, so the receiver gets it as string even we send in integer.